dll/win32/shell32/folders/CControlPanelFolder.cpp

```diff
@@ -627,11 +634,11 @@ HRESULT WINAPI CControlPanelFolder::Initialize(PCIDLIST_ABSOLUTE pidl)
     pidlRoot = ILClone(pidl);
 
     /* Create the inner reg folder */
+    REGFOLDERINITDATA RegInit = { static_cast<IShellFolder*>(this), &g_RegFolderInfo };
     HRESULT hr;
-    hr = CRegFolder_CreateInstance(&CLSID_ControlPanel,
+    hr = CRegFolder_CreateInstance(&RegInit,
                                    pidlRoot,
                                    L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}",
```

Comment:
What do these guids correspond to? Are they related somehow to ControlPanel? If so, can't they be moved as part of a field of the REGFOLDERINFO? (if that structure is your own and private to shell32 only)

Reply:
It is the parsing path of the control panel yes. I'm sure we can move more stuff into REGFOLDERINFO over time but the real change we should be making is for CRegFolder to be COM aggregated like on Windows.